crates/engine/tree/src/tree/mod.rs

```diff
@@ -2224,13 +2224,47 @@ where
 
         let exec_time = Instant::now();
 
-        // TODO: create StateRootTask with the receiving end of a channel and
-        // pass the sending end of the channel to the state hook.
-        let noop_state_hook = |_state: &EvmState| {};
+        let persistence_not_in_progress = !self.persistence_state.in_progress();
+
+        // TODO: uncomment to use StateRootTask
```

Comment:
This doesn't include the changes for using the `std::thread::scope` that spawning `StateRootTask` requires now

Reply:
yep true, but they were quite invasive because it requires a huge scope that different stuff gets in, and also a custom return from that scope. So I decided to leave it for the wiring PR.